projects/angular/src/forms/combobox/options.ts

```diff
@@ -44,7 +44,11 @@ let nbOptionsComponents = 0;
 
     <!-- Rendered if data set is empty -->
     <div *ngIf="emptyOptions">
-      <span class="clr-combobox-options-empty-text">
+      <span
+        class="clr-combobox-options-empty-text"
+        id="no-results-in-combobox"
```

Comment:
This must be a unique id because there could be more than on combobox on the page. You can use the `uniqueIdFactory` function for this.

Sorry if this isn't ready for review. It's not a draft, and I don't want this to be missed.

Reply:
Your comment is on point and it's my bad I didn't set this PR to draft.

I was actually thinking of having an id that is concatenation of the id of combobox + '-no-results', and I guess using this function makes more sense. Thanks!

On a different hand though the reality should be that we should never have two open dialogs showing "no results" at the same time (because the first one will close once we start dealing with the second combobox), but reusing id-s is not something we need and is counterintuitive

Reply:
Updated and this how id-s are now for more than one combobox in the same page:

[multiple-ids.webm](https://github.com/vmware-clarity/ng-clarity/assets/132376042/fe028d68-af49-4223-a1d3-82b2acaaff40)

Reply:
This is now "clr-id-0" and "clr-id-1", maybe a better naming would be "clr-no-results-id-1" and "clr-no-results-id-2"?

Reply:
>I was actually thinking of having an id that is concatenation of the id of the combobox + '-no-results'

That would be fine as well.

Reply:
Should we move the id of the combobox to be kept in the option-selection.service.ts then so that all id-s are in the state which this service holds? I actually think using the uniqueIdFactory function it is a simpler solution. As for the naming maybe we could have "clr-id-1-no-results" to avoid parsing of the text that the function  returns.

Reply:
We don't need the id in the service at all. The `ClrOptions` component has a `optionsId` property. We can just append to that.

Reply:
We don't need the id in the service at all. The `ClrOptions` component has a `optionsId` property. We can just append to that. See here: 1e7f2c0a4e2b89e60587113290540c2c81021dde

Reply:
It seems this solution is not full as on Mac OS the screen reader is still not announcing "No results". Once I fix it for Mac OS I will try to reuse your commit, thanks.